dataset/dataset.cpp

```diff
@@ -175,6 +148,26 @@ Dataset Dataset::slice(const Slice s) const {
   return out;
 }
 
+Dataset &Dataset::setSlice(const Slice s, const Dataset &data) {
+  // TODO Need dry-run mechanism here?
```

Comment:
Your comment here relates to partial application in case we encounter read-only problems?

Reply:
Yes, though read-only is not necessarily the only issue. It can also be a missing meta data item on the LHS.